metaflow/metaflow_git.py

```diff
@@ -17,71 +17,79 @@
 
 def _call_git(
     args: List[str], path=Union[str, os.PathLike]
-) -> subprocess.CompletedProcess[str]:
-    result = subprocess.run(
-        ["git", *args],
-        cwd=path,
-        capture_output=True,
-        text=True,
-        check=False,
-    )
-    return result
+) -> Tuple[Optional[str], Optional[int], bool]:
+    """
+    Call git with provided args.
+
+    Returns
+    -------
+        tuple : Tuple containing
+            (stdout, exitcode, failure) of the call
+    """
+    try:
+        result = subprocess.run(
+            ["git", *args],
+            cwd=path,
+            capture_output=True,
+            text=True,
+            check=False,
+        )
+        return result.stdout.strip(), result.returncode, False
```

Comment:
do we need to return Failure @saikonen ?

Reply:
the failure is a separate case here for subprocess issues (e.g. timeout) that is not captured by the process exit code. I introduced this in order to retain the previous logic with the try/except, though for the return values there is no difference and could be omitted from what I can tell.